rust/src/core.rs

```diff
@@ -295,6 +295,10 @@ impl Flow {
             let mut secs: u64 = 0;
             let mut usecs: u64 = 0;
             FlowGetLastTimeAsParts(self, &mut secs, &mut usecs);
+            if usecs > 999999 {
```

Comment:
have you observed this?

Reply:
From fuzzing yes...
Maybe I can change fuzzpcap to produce only valid timestamps...

Reply:
this comes directly from `Flow::last_ts`, which is updated directly from `Packet::timestamp` iirc, so perhaps some lower level `DEBUG_VALIDATE_BUG_ON`s might be helpful.

Reply:
This comes indeed from `flow->lastts.tv_usec` which comes from `COPY_TIMESTAMP(&p->ts, &f->lastts);`

Then, it depends on the source.

For the fuzz targets, there is `p->ts.tv_usec = header->ts.tv_usec;` where header comes from `pcap_next_ex(pkts, &header, &pkt);
Same goes for source-pcap-file-helper.c:    `p->ts.tv_usec = h->ts.tv_usec;`

Should the sources rather do `p->ts.tv_usec = header->ts.tv_usec % 1000000;` ?
and then we can have a `DEBUG_VALIDATE_BUG_ON` later for all sources ?